### ISSUE TITLE
Keyboar map

### ISSUE THREAD

**Issue by adeleglise** [tamlok/vnote#102]
Hi,

I'm currently using the latest release AppImage on ArchLinux.

The problem is I'm using a keyboard with a french mapping, and in Vnote the keyboard is mapped to english, I've looked in the settings and I can't find how to change this.

**Comment by tamlok** [comment]
Thanks for your feedback! I didn't consider the keyboard mapping before. I will take a look in it.

**Comment by tamlok** [comment]
Hi,

I have looked into the keyboard layout handling in Qt. I think it is the shortcuts in VNote that cause inconvenience in a French locale keyboard layout, is it? Everything else works as expected, right?

If that is the case, I think I need your help to translate the shortcut key sequences into French.

In VNote, shortcuts are defined in `vnote.ini` like this:

```
; Leader key of Captain mode
CaptainMode=Ctrl+E
; Create a note in current folder
NewNote=Ctrl+Alt+N
; Create a subfolder in current folder
NewSubfolder=Ctrl+Alt+S
; Save current note
SaveNote=Ctrl+S
; Save changes and enter read mode
SaveAndRead=Ctrl+T
; Edit current note
EditNote=Ctrl+W
; Close current note
CloseNote=
; Open file/replace dialog
Find=Ctrl+F
; Find next occurence
FindNext=F3
; Find previous occurence
FindPrevious=Shift+F3
; Recover last closed file
LastClosedFile=Ctrl+Shift+T
; Activate next tab
ActivateNextTab=Ctrl+Tab
; Activate previous tab
ActivatePreviousTab=Ctrl+Shift+Tab
; Activate flash page
FlashPage=Ctrl+Alt+L
; Open via system's default program
OpenViaDefaultProgram=F12
```

We need to translate those key sequences which have problems in French mapping. So could you give out a concrete example?

Thanks!

**Comment by adeleglise** [comment]
Hey sorry for the delay. I'll have a look at this and come back to you ASAP !

**Comment by adeleglise** [comment]
So, after testing, I can confirm that only the shortcuts are affected by the keyboard mapping.

When I try to go in edit mode Ctrl+W, it undo the last action (Ctrl+Z).

**Comment by tamlok** [comment]
Are you using AZERTY keyboard like this:

![AZERTY](https://upload.wikimedia.org/wikipedia/commons/thumb/b/b9/KB_France.svg/750px-KB_France.svg.png)

If yes, you hit `Ctrl` and `W` at the bottom-left corner when you mean `Ctrl+W`, right? And this will trigger the UNDO action? And if you hit `Ctrl` and `Z` at the up-left corner, you will trigger the edit mode?

Thanks!

**Comment by adeleglise** [comment]
The behaviour is really odd. If i'm not in vim mode, open a new note and hit Ctrl+W I enter edit mode, if I hit Ctrl+W again it starts to erase text as an undo command.

I think that the two actions are trigered at the same time, "undo" and "enter edit mode".

**Comment by tamlok** [comment]
Oh, that is by design. Ctrl+w will trigger edit mode if you are now in read mode. It will delete a word if you are in edit mode, which is the behavior of Vim and bash shell.

Ctrl+T will save changes and exit edit mode. You could reference the shortcuts help in the help menu in detail.

Thanks!

**Comment by adeleglise** [comment]
Ok, i think there's no real issue then ^^

I'll keep you in touch if I encounter something else.

**Comment by Matows** [comment]
Hello!
I just tried VNote and it seems fine but it's not usable for azerty keyboard :
Everywhere in the software, the keyboard layout is qwerty. The section concerning layouts in settings isn't applying (I've remaped everything I could). And some keys are missing (lower/uppercase, =, +, !, : etc.). All of those keys are placed differently compared to qwerty layout.
That said, that's the only reason I can't use it. I'm looking forward to use it !
